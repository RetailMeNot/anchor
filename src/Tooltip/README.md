# Tooltip

The Tooltip component is a wrapper for any element that needs a tooltip that displays to the user on hover.

## Usage

#### Basic
```jsx
import { Tooltip } from '@retailmenot/anchor';

const basicExample = props => (
  <Tooltip content="Text to show on hover">
    <button>Hover Over Me</button>
  </Tooltip>
);
```

#### Advanced
```jsx
import { Tooltip } from '@retailmenot/anchor';

const advancedExample = props => (
    <Tooltip content="Text to show on hover" position="rightEnd">
      <button>Hover Over Me</button>
    </Tooltip>
);
```

## Props

  * **`position = 'topEnd'`**: Property for the position of the Tooltip.  
  * **`maxWidth = 'auto'`:** Property for the maximum width of the Tooltip.
  * **`wrap`:** Property to control the text wrap of the Tooltip.
  * **`background = 'rgba(0, 0, 0, 0.8)`:** CSS property for the background.
  * **`color = 'white'`:** CSS property for the text color.
  * **`content`:** Property for the text of the Tooltip.

