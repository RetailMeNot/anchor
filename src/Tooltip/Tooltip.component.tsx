// REACT
import * as React from 'react';
// VENDOR
import classNames from 'classnames';
import styled, { css } from '@xstyled/styled-components';
import { th } from '@xstyled/system';

export type Position =
    | 'topStart'
    | 'top'
    | 'topEnd'
    | 'rightStart'
    | 'right'
    | 'rightEnd'
    | 'bottomEnd'
    | 'bottom'
    | 'bottomStart'
    | 'leftEnd'
    | 'left'
    | 'leftStart';

interface TooltipContainerProps extends React.HTMLAttributes<HTMLDivElement> {
    content: string;
    position?: Position;
    maxWidth?: string;
    background?: string;
    color?: string;
    wrap?: boolean;
    display?: string;
}

interface TooltipContainerState {
    width: number;
    height: number;
    toolTipHeight: number;
    toolTipWidth: number;
    hidden: boolean;
}

const ToolTipContainer = styled('div')<TooltipContainerProps>`
    position: relative;
    ${({ display = 'inline-block' }) => css({ display })};
`;

interface TooltipElementProps {
    position: Position;
    height: number;
    width: number;
    toolTipHeight: number;
    toolTipWidth: number;
    wrap?: boolean;
    background?: string;
    color?: string;
    maxWidth?: string;
}

const tooltipMargin = 8;

const positionVariants = (
    position: Position,
    height: number,
    width: number,
    toolTipHeight: number,
    toolTipWidth: number
) => {
    switch (position) {
        case 'topStart':
            return css({
                left: 0,
                top: 0,
                transform: `translate3d(0, -${toolTipHeight +
                    tooltipMargin}px, 0)`,
            });
        case 'top':
            return css({
                left: 0,
                top: 0,
                transform: `translate3d(${width / 2 -
                    toolTipWidth / 2}px, -${toolTipHeight +
                    tooltipMargin}px, 0)`,
            });
        case 'topEnd':
            return css({
                right: 0,
                top: 0,
                transform: `translate3d(0, -${toolTipHeight +
                    tooltipMargin}px, 0)`,
            });
        case 'rightStart':
            return css({
                left: 0,
                top: 0,
                transform: `translate3d(${width + tooltipMargin}px, 0, 0)`,
            });
        case 'right':
            return css({
                left: 0,
                top: 0,
                transform: `translate3d(${width + tooltipMargin}px, ${height /
                    2 -
                    toolTipHeight / 2}px, 0)`,
            });
        case 'rightEnd':
            return css({
                left: 0,
                bottom: 0,
                transform: `translate3d(${width + tooltipMargin}px, 0, 0)`,
            });
        case 'bottomEnd':
            return css({
                right: 0,
                top: 0,
                transform: `translate3d(0, ${height + tooltipMargin}px, 0)`,
            });
        case 'bottom':
            return css({
                left: 0,
                top: 0,
                transform: `translate3d(${width / 2 -
                    toolTipWidth / 2}px, ${height + tooltipMargin}px, 0)`,
            });
        case 'bottomStart':
            return css({
                left: 0,
                top: 0,
                transform: `translate3d(0, ${height + tooltipMargin}px, 0)`,
            });
        case 'leftEnd':
            return css({
                left: 0,
                bottom: 0,
                transform: `translate3d(-${toolTipWidth +
                    tooltipMargin}px, 0, 0)`,
            });
        case 'left':
            return css({
                left: 0,
                top: 0,
                transform: `translate3d(-${toolTipWidth +
                    tooltipMargin}px, ${height / 2 - toolTipHeight / 2}px, 0)`,
            });
        case 'leftStart':
            return css({
                left: 0,
                top: 0,
                transform: `translate3d(-${toolTipWidth +
                    tooltipMargin}px, 0, 0)`,
            });
        default:
            return css({
                right: 0,
                top: 0,
                transform: `translate3d(0, -${height + tooltipMargin}px, 0)`,
            });
    }
};

const TooltipElement = styled('div')<TooltipElementProps>`
    position: absolute;
    box-sizing: border-box;
    ${({ background = 'rgba(0, 0, 0, 0.8)', color = 'white' }) =>
        css({ background, color })};
    border-radius: base;
    font-family: ${th('typography.fontFamily')};
    font-size: 0.8rem;
    padding: 0.5rem;
    opacity: 0;
    visibility: hidden;
    transition: opacity 250ms ease-in-out;

    &.active {
        opacity: 1;
        visibility: visible;
    }

    ${({ wrap = true }) =>
        wrap ? css({ whiteSpace: 'normal' }) : css({ whiteSpace: 'nowrap' })};

    ${({ maxWidth = 'auto' }) => css({ width: maxWidth })};

    ${({ position, height, width, toolTipHeight, toolTipWidth }) =>
        positionVariants(position, height, width, toolTipHeight, toolTipWidth)};
`;

export class Tooltip extends React.Component<
    TooltipContainerProps,
    TooltipContainerState
> {
    public state = {
        width: 0,
        height: 0,
        toolTipHeight: 0,
        toolTipWidth: 0,
        hidden: true,
    };
    private readonly tooltipContainerRef: React.RefObject<
        HTMLDivElement
    > = React.createRef();
    private readonly tooltipRef: React.RefObject<
        HTMLDivElement
    > = React.createRef();

    componentDidMount(): void {
        const { current: toolTipContainer } = this.tooltipContainerRef;
        const { current: toolTip } = this.tooltipRef;
        // TODO: instead of setting state, what about using the render fxn?
        if (toolTipContainer) {
            this.setState({
                height: toolTipContainer.clientHeight,
                width: toolTipContainer.clientWidth,
            });
        }
        if (toolTip) {
            this.setState({
                toolTipHeight: toolTip.clientHeight,
                toolTipWidth: toolTip.clientWidth,
            });
        }
    }

    render(): React.ReactElement<Tooltip> {
        const {
            className,
            children,
            content,
            position,
            wrap,
            background,
            color,
            maxWidth,
            ...props
        } = this.props;
        const { height, width, toolTipHeight, toolTipWidth } = this.state;
        return (
            <ToolTipContainer
                content={content}
                onMouseEnter={() => this.setState({ hidden: false })}
                onMouseLeave={() => this.setState({ hidden: true })}
                ref={this.tooltipContainerRef}
                className={classNames('anchor-tooltip', className)}
                {...props}
            >
                {children}
                <TooltipElement
                    className={classNames('anchor-tooltip-element', {
                        active: !this.state.hidden,
                    })}
                    ref={this.tooltipRef}
                    position={position || 'topEnd'}
                    height={height}
                    width={width}
                    toolTipHeight={toolTipHeight}
                    toolTipWidth={toolTipWidth}
                    children={content}
                    wrap={wrap}
                    background={background}
                    color={color}
                    maxWidth={maxWidth}
                />
            </ToolTipContainer>
        );
    }
}
