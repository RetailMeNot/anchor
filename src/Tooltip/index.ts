export { Tooltip } from './Tooltip.component';
