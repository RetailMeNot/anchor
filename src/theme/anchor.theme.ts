// ANCHOR THEME
import { radii } from './sizes.theme';
import { MODAL_THEME_KEY, MODAL_THEME } from '../Modal';

// Copied from Adapter.StandardBreakpoints mins
// todo: dedup- they should use a single source
const breakpoints = {
    sm: '576px',
    md: '768px',
    lg: '992px',
    xl: '1200px',
};

interface AnchorTheme {
    colors?: any;
    breakpoints?: any;
    radii?: any;
    [key: string]: any;
}

export const anchorTheme: AnchorTheme = {
    breakpoints,
    radii,
    [MODAL_THEME_KEY]: MODAL_THEME,
};
