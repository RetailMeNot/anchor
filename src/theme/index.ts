export { default as colors } from './colors.theme';
export { default as fonts } from './fonts.theme';
export { default as sizes } from './sizes.theme';
export { default as variables } from './variables.theme';

export { NormalizeCSS, GlobalCSS } from './GlobalStyles';
