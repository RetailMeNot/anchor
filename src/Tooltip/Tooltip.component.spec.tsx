// REACT
import * as React from 'react';
import * as renderer from 'react-test-renderer';
// VENDOR
import { ThemeProvider } from '@xstyled/styled-components';
// COMPONENT
import { RootTheme } from '../theme';
import { Tooltip } from './Tooltip.component';
import { Position } from '../utils/position/position';

// ENZYME
import { mount, shallow } from 'enzyme';

const subject = (
    <ThemeProvider theme={RootTheme}>
        <Tooltip content="This is a tooltip">
            <button>Hover</button>
        </Tooltip>
    </ThemeProvider>
);

describe('Component: Tooltip', () => {
    it('should be defined and match its snapshot.', () => {
        const component = shallow(subject);
        const tree = renderer.create(subject).toJSON();

        expect(subject).toBeDefined();
        expect(component).toBeDefined();
        expect(tree).toMatchSnapshot();
    });
    it('should show a tooltip on hover.', () => {
        const wrapper = mount(subject);
        expect(wrapper.find(Tooltip).state().hidden).toBeTruthy();
        wrapper.find(Tooltip).simulate('mouseenter');
        wrapper.update();
        expect(wrapper.find(Tooltip).state().hidden).toBeFalsy();
        wrapper.find(Tooltip).simulate('mouseleave');
        wrapper.update();
        expect(wrapper.find(Tooltip).state().hidden).toBeTruthy();
    });
    it("should force tooltip's visibility when show is set to true", () => {
        const testSubject = (
            <ThemeProvider theme={RootTheme}>
                <Tooltip content="This is a tooltip" show={true}>
                    <button>Hover</button>
                </Tooltip>
            </ThemeProvider>
        );

        const wrapper = mount(testSubject);

        wrapper.simulate('mouseleave');
        wrapper.update();

        expect(
            wrapper.find('.anchor-tooltip-element.active').exists()
        ).toBeTruthy();
    });
    it('should show an arrow when showArrow is true.', () => {
        const testSubject = (
            <ThemeProvider theme={RootTheme}>
                <Tooltip
                    content="This is a tooltip"
                    show={true}
                    showArrow={true}
                >
                    <button>Hover</button>
                </Tooltip>
            </ThemeProvider>
        );

        const wrapper = mount(testSubject);

        expect(wrapper.find('Arrow').exists()).toBeTruthy();
    });
    it('should render all the different tooltip positions.', () => {
        [
            'topStart',
            'top',
            'topEnd',
            'rightStart',
            'right',
            'rightEnd',
            'bottomEnd',
            'bottom',
            'bottomStart',
            'leftEnd',
            'left',
            'leftStart',
        ].map((option: Position) => {
            const subjectInstance = (
                <ThemeProvider theme={RootTheme}>
                    <Tooltip content="This is a tooltip" position={option}>
                        <button>Hover</button>
                    </Tooltip>
                </ThemeProvider>
            );
            const component = shallow(subjectInstance);
            const tree = renderer.create(subjectInstance).toJSON();

            expect(subjectInstance).toBeDefined();
            expect(component).toBeDefined();
            expect(tree).toMatchSnapshot();
        });
    });
});
