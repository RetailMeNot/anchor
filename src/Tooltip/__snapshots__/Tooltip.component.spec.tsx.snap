// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Tooltip should be defined and match its snapshot. 1`] = `
.c1 {
  position: absolute;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 1;
  min-width: 10rem;
  opacity: 0;
  visibility: hidden;
  -webkit-transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  color: white;
  padding: 0.5rem;
  border-radius: 4px;
  right: 0;
  top: 0;
  -webkit-transform: translate3d(0,-8px,0);
  -ms-transform: translate3d(0,-8px,0);
  transform: translate3d(0,-8px,0);
  white-space: normal;
  width: auto;
}

.c1.active {
  opacity: 1;
  visibility: visible;
}

.c2 {
  position: absolute;
  box-sizing: border-box;
  border-radius: inherit;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: black;
}

.c3 {
  position: relative;
  z-index: 1;
}

.c0 {
  position: relative;
  display: inline-block;
}

<div
  className="c0 anchor-tooltip"
  content="This is a tooltip"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <button>
    Hover
  </button>
  <div
    className="c1 anchor-tooltip-element"
    color="white"
    height={0}
    spacing={8}
    width={0}
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    >
      This is a tooltip
    </div>
  </div>
</div>
`;

exports[`Component: Tooltip should render all the different tooltip positions. 1`] = `
.c1 {
  position: absolute;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 1;
  min-width: 10rem;
  opacity: 0;
  visibility: hidden;
  -webkit-transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  color: white;
  padding: 0.5rem;
  border-radius: 4px;
  left: 0;
  top: 0;
  -webkit-transform: translate3d(0,-8px,0);
  -ms-transform: translate3d(0,-8px,0);
  transform: translate3d(0,-8px,0);
  white-space: normal;
  width: auto;
}

.c1.active {
  opacity: 1;
  visibility: visible;
}

.c2 {
  position: absolute;
  box-sizing: border-box;
  border-radius: inherit;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: black;
}

.c3 {
  position: relative;
  z-index: 1;
}

.c0 {
  position: relative;
  display: inline-block;
}

<div
  className="c0 anchor-tooltip"
  content="This is a tooltip"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <button>
    Hover
  </button>
  <div
    className="c1 anchor-tooltip-element"
    color="white"
    height={0}
    spacing={8}
    width={0}
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    >
      This is a tooltip
    </div>
  </div>
</div>
`;

exports[`Component: Tooltip should render all the different tooltip positions. 2`] = `
.c1 {
  position: absolute;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 1;
  min-width: 10rem;
  opacity: 0;
  visibility: hidden;
  -webkit-transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  color: white;
  padding: 0.5rem;
  border-radius: 4px;
  left: 0;
  top: 0;
  -webkit-transform: translate3d(0px,-8px,0);
  -ms-transform: translate3d(0px,-8px,0);
  transform: translate3d(0px,-8px,0);
  white-space: normal;
  width: auto;
}

.c1.active {
  opacity: 1;
  visibility: visible;
}

.c2 {
  position: absolute;
  box-sizing: border-box;
  border-radius: inherit;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: black;
}

.c3 {
  position: relative;
  z-index: 1;
}

.c0 {
  position: relative;
  display: inline-block;
}

<div
  className="c0 anchor-tooltip"
  content="This is a tooltip"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <button>
    Hover
  </button>
  <div
    className="c1 anchor-tooltip-element"
    color="white"
    height={0}
    spacing={8}
    width={0}
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    >
      This is a tooltip
    </div>
  </div>
</div>
`;

exports[`Component: Tooltip should render all the different tooltip positions. 3`] = `
.c1 {
  position: absolute;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 1;
  min-width: 10rem;
  opacity: 0;
  visibility: hidden;
  -webkit-transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  color: white;
  padding: 0.5rem;
  border-radius: 4px;
  right: 0;
  top: 0;
  -webkit-transform: translate3d(0,-8px,0);
  -ms-transform: translate3d(0,-8px,0);
  transform: translate3d(0,-8px,0);
  white-space: normal;
  width: auto;
}

.c1.active {
  opacity: 1;
  visibility: visible;
}

.c2 {
  position: absolute;
  box-sizing: border-box;
  border-radius: inherit;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: black;
}

.c3 {
  position: relative;
  z-index: 1;
}

.c0 {
  position: relative;
  display: inline-block;
}

<div
  className="c0 anchor-tooltip"
  content="This is a tooltip"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <button>
    Hover
  </button>
  <div
    className="c1 anchor-tooltip-element"
    color="white"
    height={0}
    spacing={8}
    width={0}
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    >
      This is a tooltip
    </div>
  </div>
</div>
`;

exports[`Component: Tooltip should render all the different tooltip positions. 4`] = `
.c1 {
  position: absolute;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 1;
  min-width: 10rem;
  opacity: 0;
  visibility: hidden;
  -webkit-transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  color: white;
  padding: 0.5rem;
  border-radius: 4px;
  left: 0;
  top: 0;
  -webkit-transform: translate3d(8px,0,0);
  -ms-transform: translate3d(8px,0,0);
  transform: translate3d(8px,0,0);
  white-space: normal;
  width: auto;
}

.c1.active {
  opacity: 1;
  visibility: visible;
}

.c2 {
  position: absolute;
  box-sizing: border-box;
  border-radius: inherit;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: black;
}

.c3 {
  position: relative;
  z-index: 1;
}

.c0 {
  position: relative;
  display: inline-block;
}

<div
  className="c0 anchor-tooltip"
  content="This is a tooltip"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <button>
    Hover
  </button>
  <div
    className="c1 anchor-tooltip-element"
    color="white"
    height={0}
    spacing={8}
    width={0}
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    >
      This is a tooltip
    </div>
  </div>
</div>
`;

exports[`Component: Tooltip should render all the different tooltip positions. 5`] = `
.c1 {
  position: absolute;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 1;
  min-width: 10rem;
  opacity: 0;
  visibility: hidden;
  -webkit-transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  color: white;
  padding: 0.5rem;
  border-radius: 4px;
  left: 0;
  top: 0;
  -webkit-transform: translate3d(8px,0px,0);
  -ms-transform: translate3d(8px,0px,0);
  transform: translate3d(8px,0px,0);
  white-space: normal;
  width: auto;
}

.c1.active {
  opacity: 1;
  visibility: visible;
}

.c2 {
  position: absolute;
  box-sizing: border-box;
  border-radius: inherit;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: black;
}

.c3 {
  position: relative;
  z-index: 1;
}

.c0 {
  position: relative;
  display: inline-block;
}

<div
  className="c0 anchor-tooltip"
  content="This is a tooltip"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <button>
    Hover
  </button>
  <div
    className="c1 anchor-tooltip-element"
    color="white"
    height={0}
    spacing={8}
    width={0}
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    >
      This is a tooltip
    </div>
  </div>
</div>
`;

exports[`Component: Tooltip should render all the different tooltip positions. 6`] = `
.c1 {
  position: absolute;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 1;
  min-width: 10rem;
  opacity: 0;
  visibility: hidden;
  -webkit-transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  color: white;
  padding: 0.5rem;
  border-radius: 4px;
  left: 0;
  bottom: 0;
  -webkit-transform: translate3d(8px,0,0);
  -ms-transform: translate3d(8px,0,0);
  transform: translate3d(8px,0,0);
  white-space: normal;
  width: auto;
}

.c1.active {
  opacity: 1;
  visibility: visible;
}

.c2 {
  position: absolute;
  box-sizing: border-box;
  border-radius: inherit;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: black;
}

.c3 {
  position: relative;
  z-index: 1;
}

.c0 {
  position: relative;
  display: inline-block;
}

<div
  className="c0 anchor-tooltip"
  content="This is a tooltip"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <button>
    Hover
  </button>
  <div
    className="c1 anchor-tooltip-element"
    color="white"
    height={0}
    spacing={8}
    width={0}
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    >
      This is a tooltip
    </div>
  </div>
</div>
`;

exports[`Component: Tooltip should render all the different tooltip positions. 7`] = `
.c1 {
  position: absolute;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 1;
  min-width: 10rem;
  opacity: 0;
  visibility: hidden;
  -webkit-transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  color: white;
  padding: 0.5rem;
  border-radius: 4px;
  right: 0;
  top: 0;
  -webkit-transform: translate3d(0,8px,0);
  -ms-transform: translate3d(0,8px,0);
  transform: translate3d(0,8px,0);
  white-space: normal;
  width: auto;
}

.c1.active {
  opacity: 1;
  visibility: visible;
}

.c2 {
  position: absolute;
  box-sizing: border-box;
  border-radius: inherit;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: black;
}

.c3 {
  position: relative;
  z-index: 1;
}

.c0 {
  position: relative;
  display: inline-block;
}

<div
  className="c0 anchor-tooltip"
  content="This is a tooltip"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <button>
    Hover
  </button>
  <div
    className="c1 anchor-tooltip-element"
    color="white"
    height={0}
    spacing={8}
    width={0}
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    >
      This is a tooltip
    </div>
  </div>
</div>
`;

exports[`Component: Tooltip should render all the different tooltip positions. 8`] = `
.c1 {
  position: absolute;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 1;
  min-width: 10rem;
  opacity: 0;
  visibility: hidden;
  -webkit-transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  color: white;
  padding: 0.5rem;
  border-radius: 4px;
  left: 0;
  top: 0;
  -webkit-transform: translate3d(0px,8px,0);
  -ms-transform: translate3d(0px,8px,0);
  transform: translate3d(0px,8px,0);
  white-space: normal;
  width: auto;
}

.c1.active {
  opacity: 1;
  visibility: visible;
}

.c2 {
  position: absolute;
  box-sizing: border-box;
  border-radius: inherit;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: black;
}

.c3 {
  position: relative;
  z-index: 1;
}

.c0 {
  position: relative;
  display: inline-block;
}

<div
  className="c0 anchor-tooltip"
  content="This is a tooltip"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <button>
    Hover
  </button>
  <div
    className="c1 anchor-tooltip-element"
    color="white"
    height={0}
    spacing={8}
    width={0}
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    >
      This is a tooltip
    </div>
  </div>
</div>
`;

exports[`Component: Tooltip should render all the different tooltip positions. 9`] = `
.c1 {
  position: absolute;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 1;
  min-width: 10rem;
  opacity: 0;
  visibility: hidden;
  -webkit-transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  color: white;
  padding: 0.5rem;
  border-radius: 4px;
  left: 0;
  top: 0;
  -webkit-transform: translate3d(0,8px,0);
  -ms-transform: translate3d(0,8px,0);
  transform: translate3d(0,8px,0);
  white-space: normal;
  width: auto;
}

.c1.active {
  opacity: 1;
  visibility: visible;
}

.c2 {
  position: absolute;
  box-sizing: border-box;
  border-radius: inherit;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: black;
}

.c3 {
  position: relative;
  z-index: 1;
}

.c0 {
  position: relative;
  display: inline-block;
}

<div
  className="c0 anchor-tooltip"
  content="This is a tooltip"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <button>
    Hover
  </button>
  <div
    className="c1 anchor-tooltip-element"
    color="white"
    height={0}
    spacing={8}
    width={0}
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    >
      This is a tooltip
    </div>
  </div>
</div>
`;

exports[`Component: Tooltip should render all the different tooltip positions. 10`] = `
.c1 {
  position: absolute;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 1;
  min-width: 10rem;
  opacity: 0;
  visibility: hidden;
  -webkit-transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  color: white;
  padding: 0.5rem;
  border-radius: 4px;
  left: 0;
  bottom: 0;
  -webkit-transform: translate3d(-8px,0,0);
  -ms-transform: translate3d(-8px,0,0);
  transform: translate3d(-8px,0,0);
  white-space: normal;
  width: auto;
}

.c1.active {
  opacity: 1;
  visibility: visible;
}

.c2 {
  position: absolute;
  box-sizing: border-box;
  border-radius: inherit;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: black;
}

.c3 {
  position: relative;
  z-index: 1;
}

.c0 {
  position: relative;
  display: inline-block;
}

<div
  className="c0 anchor-tooltip"
  content="This is a tooltip"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <button>
    Hover
  </button>
  <div
    className="c1 anchor-tooltip-element"
    color="white"
    height={0}
    spacing={8}
    width={0}
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    >
      This is a tooltip
    </div>
  </div>
</div>
`;

exports[`Component: Tooltip should render all the different tooltip positions. 11`] = `
.c1 {
  position: absolute;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 1;
  min-width: 10rem;
  opacity: 0;
  visibility: hidden;
  -webkit-transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  color: white;
  padding: 0.5rem;
  border-radius: 4px;
  left: 0;
  top: 0;
  -webkit-transform: translate3d(-8px,0px,0);
  -ms-transform: translate3d(-8px,0px,0);
  transform: translate3d(-8px,0px,0);
  white-space: normal;
  width: auto;
}

.c1.active {
  opacity: 1;
  visibility: visible;
}

.c2 {
  position: absolute;
  box-sizing: border-box;
  border-radius: inherit;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: black;
}

.c3 {
  position: relative;
  z-index: 1;
}

.c0 {
  position: relative;
  display: inline-block;
}

<div
  className="c0 anchor-tooltip"
  content="This is a tooltip"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <button>
    Hover
  </button>
  <div
    className="c1 anchor-tooltip-element"
    color="white"
    height={0}
    spacing={8}
    width={0}
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    >
      This is a tooltip
    </div>
  </div>
</div>
`;

exports[`Component: Tooltip should render all the different tooltip positions. 12`] = `
.c1 {
  position: absolute;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 1;
  min-width: 10rem;
  opacity: 0;
  visibility: hidden;
  -webkit-transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  transition: visibility 250ms ease-in-out,opacity 250ms ease-in-out;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  color: white;
  padding: 0.5rem;
  border-radius: 4px;
  left: 0;
  top: 0;
  -webkit-transform: translate3d(-8px,0,0);
  -ms-transform: translate3d(-8px,0,0);
  transform: translate3d(-8px,0,0);
  white-space: normal;
  width: auto;
}

.c1.active {
  opacity: 1;
  visibility: visible;
}

.c2 {
  position: absolute;
  box-sizing: border-box;
  border-radius: inherit;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: black;
}

.c3 {
  position: relative;
  z-index: 1;
}

.c0 {
  position: relative;
  display: inline-block;
}

<div
  className="c0 anchor-tooltip"
  content="This is a tooltip"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <button>
    Hover
  </button>
  <div
    className="c1 anchor-tooltip-element"
    color="white"
    height={0}
    spacing={8}
    width={0}
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    >
      This is a tooltip
    </div>
  </div>
</div>
`;
